### PR TITLE
ENH: Added optional pickle protocol version argument to pandas.to_pickle()

### LINKED ISSUE #14488: Adding optional pickle protocol version argument to pandas.to_pickle()
Sometimes pickle files saved in Python v3.x are needed to read in Python v2.x. It would be nice if one can easily set pickle protocol version in `to_pickle()`.

It can be done with the following little changes:

In file `/pandas/io/pickle.py`:

```
def to_pickle(obj, path, protocol=pkl.HIGHEST_PROTOCOL):
    """
    Pickle (serialize) object to input file path

    Parameters
    ----------
    obj : any object
    path : string
        File path
    """
    with open(path, 'wb') as f:
        pkl.dump(obj, f, protocol=protocol)
```

In file `/pandas/core/generic.py`:

```
    def to_pickle(self, path, protocol):
        """
        Pickle (serialize) object to input file path.

        Parameters
        ----------
        path : string
            File path
        """
        from pandas.io.pickle import to_pickle
        return to_pickle(self, path, protocol)
```

### PR DESCRIPTION
**Pull request by josephbwagner**
- [ ] closes #14488 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

… :issue:'14488'

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1200,7 +1200,7 @@ def to_sql(self, name, con, flavor=None, schema=None, if_exists='fail',
                    if_exists=if_exists, index=index, index_label=index_label,
                    chunksize=chunksize, dtype=dtype)
 
-    def to_pickle(self, path):
+    def to_pickle(self, path, protocol):
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: `protocol` should be an optional argument here. You'll probably need to have the default be `None`, and then determine the correct protocol based on the python version (see `pandas.compat`). The default should be the same as python's default.

Comment (TomAugspurger) [inline pandas/core/generic.py]: Actually HIGHEST_PROTOCOL should work I think.

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
closing as stale, but if you would like to update, pls comment.

**Comment by jbschiratti** [comment]
If the original author @beverast lost interest in the PR, I would like to take over.
I am also interested in having an optional parameter `protocol` to the `to_pickle` function.

**Comment by jreback** [comment]
sure u can start with the commits here if u like 
open a new PR

**Pull request closed without merging**